File: src/test/java/net/slipp/support/security/MockUserDetailsService.java
package net.slipp.support.security;

import java.util.ArrayList;
import java.util.List;

import net.slipp.domain.user.SocialUser;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.authority.SimpleGrantedAuthority;
import org.springframework.security.core.userdetails.User;
import org.springframework.security.core.userdetails.UserDetails;
import org.springframework.security.core.userdetails.UserDetailsService;
import org.springframework.security.core.userdetails.UsernameNotFoundException;

public abstract class MockUserDetailsService implements UserDetailsService {
    private static Logger log = LoggerFactory.getLogger(MockUserDetailsService.class);

    @Override
    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
        SocialUser socialUser = createSocialUser();
        return new User(username, socialUser.getPassword(), true, true, true, true, createGrantedAuthorities(username));  
    }
    
    protected abstract SocialUser createSocialUser();
    
    private List<GrantedAuthority> createGrantedAuthorities(String username) {
        log.debug("UserName : {}", username);
        List<GrantedAuthority> grantedAuthorities = new ArrayList<GrantedAuthority>();
        grantedAuthorities.add(new SimpleGrantedAuthority("ROLE_USER"));
        return grantedAuthorities;
    }
}

File: src/test/scala/net/slipp/support/security/MockUserDetailsService.scala
package net.slipp.support.security

import java.util.{ArrayList, List}

import net.slipp.domain.user.SocialUser
import org.slf4j.{Logger, LoggerFactory}
import org.springframework.security.core.GrantedAuthority
import org.springframework.security.core.authority.SimpleGrantedAuthority
import org.springframework.security.core.userdetails.{User, UserDetails, UserDetailsService}

abstract class MockUserDetailsService extends UserDetailsService {
  private var log: Logger = LoggerFactory.getLogger(classOf[MockUserDetailsService])

  def loadUserByUsername(username: String): UserDetails = {
    val socialUser: SocialUser = createSocialUser
    return new User(username, socialUser.getPassword, true, true, true, true, createGrantedAuthorities(username))
  }

  protected def createSocialUser: SocialUser

  private def createGrantedAuthorities(username: String): List[GrantedAuthority] = {
    log.debug("UserName : {}", username)
    val grantedAuthorities: List[GrantedAuthority] = new ArrayList[GrantedAuthority]
    grantedAuthorities.add(new SimpleGrantedAuthority("ROLE_USER"))
    return grantedAuthorities
  }
}
